src/server/i18n/index.js

```diff
@@ -17,12 +17,42 @@
  * under the License.
  */
 
+import { resolve } from 'path';
+import globby from 'globby';
 import { i18n, i18nLoader } from '@kbn/i18n';
 
+import { fromRoot } from '../../utils';
+
 export async function i18nMixin(kbnServer, server, config) {
-  const { translationPaths = [] } = kbnServer.uiExports;
   const locale = config.get('i18n.locale');
 
+  const translationsDirs = [fromRoot('src/ui/translations')];
+
+  const groupedEntries = await Promise.all([
+    ...config.get('plugins.scanDirs').map(async path => {
+      const entries = await globby('*/translations/*.json', {
+        cwd: path,
+      });
+      return entries.map(entry => resolve(path, entry));
+    }),
+
+    ...config.get('plugins.paths').map(async path => {
+      const entries = await globby('plugins/*/translations/*.json', {
```

Comment:
Paths in `plugins.path` are supposed to point to a specific plugin, so this should also include `translations/*.json` I think.

Reply:
@spalger https://github.com/elastic/kibana/blob/master/src/cli/serve/serve.js#L126
It points to`./x-pack/` folder.